src/RewardTokenHolder.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module RewardTokenHolder (
+  prewardTokenHolder,
+  pmintRewardTokenHolder,
+) where
+
+import Conversions (pconvert)
+import Plutarch.Api.V1.Value (pnormalize)
+import Plutarch.Api.V2 (
+  PCurrencySymbol,
+  PMintingPolicy,
+  PScriptContext,
+  PScriptPurpose (PMinting, PSpending),
+  PTxOutRef,
+  PValidator,
+ )
+import Plutarch.Prelude
+import Types.Constants (rewardTokenHolderTN)
+import Utils (pand'List, pfindCurrencySymbolsByTokenName, phasInput, pheadSingleton, ptryLookupValue, ptryOwnInput)
+import "liqwid-plutarch-extra" Plutarch.Extra.TermCont (
+  pletC,
+  pletFieldsC,
+  pmatchC,
+ )
+
+data PTokenHolderMintAct (s :: S)
+  = PMintHolder (Term s (PDataRecord '[]))
+  | PBurnHolder (Term s (PDataRecord '[]))
+  deriving stock (Generic)
+  deriving anyclass (PlutusType, PIsData)
+
+instance DerivePlutusType PTokenHolderMintAct where
+  type DPTStrat _ = PlutusTypeData
+
+instance PTryFrom PData PTokenHolderMintAct
+
+instance PTryFrom PData (PAsData PTokenHolderMintAct)
+
+pmintRewardTokenHolder :: Term s (PTxOutRef :--> PMintingPolicy)
```

Comment:
What is the purpose of this MintingPolicy?

Reply:
This policy along with prewardTokenHolder validator enables locking of rewards. The utxo is spent when the rewardFold is initiated after the commit fold.